src/Views/DiskView.vala

```diff
@@ -108,64 +109,63 @@ public class Installer.DiskView : AbstractInstallerView {
-                }
-            } else if (disk.is_rotational ()) {
-                icon_name = "drive-harddisk-scsi";
-            } else {
-                icon_name = "drive-harddisk-solidstate";
-            }
+        if (install_options == null) {
+            critical (_("unable to get installation options"));
+            return;
+        }
 
-            string label;
+        foreach (unowned Distinst.EraseOption disk in install_options.get_erase_options ()) {
+            var size = disk.get_sectors () * 512;
             string model = Utils.string_from_utf8 (disk.get_model ());
-            if (model.length == 0) {
-                label = Utils.string_from_utf8 (disk.get_serial ()).replace ("_", " ");
-            } else {
-                label = model;
-            }
-
             string path = Utils.string_from_utf8 (disk.get_device_path ());
+            string icon_name = Utils.string_from_utf8 (disk.get_linux_icon ());
 
             var disk_button = new DiskButton (
-                label,
+                model,
                 icon_name,
                 path,
                 size
             );
 
-            if (size < minimum_disk_size) {
-                disk_button.set_sensitive (false);
-
-                disabled_buttons += disk_button;
-            } else {
+            if (disk.meets_requirements ()) {
                 disk_button.clicked.connect (() => {
                     if (disk_button.active) {
                         disk_grid.get_children ().foreach ((child) => {
                             ((Gtk.ToggleButton)child).active = child == disk_button;
                         });
 
+                        var opts = InstallOptions.get_default ();
+
+                        if (opts.has_recovery ()) {
+                            unowned Distinst.InstallOptions options = opts.get_options ();
+                            var recovery = options.get_recovery_option ();
+
+                            InstallOptions.get_default ().selected_option = new Distinst.InstallOption () {
+                                tag = Distinst.InstallOptionVariant.RECOVERY,
+                                option = (void*) recovery,
+                                encrypt_pass = null
+                            };
+                        } else {
+                            InstallOptions.get_default ().selected_option = new Distinst.InstallOption () {
+                                tag = Distinst.InstallOptionVariant.ERASE,
+                                option = (void*) disk,
+                                encrypt_pass = null
+                            };
+                        }
+
                         next_button.sensitive = true;
                     } else {
                         next_button.sensitive = false;
                     }
                 });
 
                 enabled_buttons += disk_button;
+            } else {
+                disk_button.set_sensitive (false);
```

Comment:
Should probably be `.sensitive = false` like the others

Reply:
Seems like `vala-lint` should check for these.